dist/js/multiselect.js

```diff
@@ -1,755 +1,1812 @@
+             */
+            function initialOrderComparison(a, b) {
+                /** The initial index for a
+                 * @type {number} */
+                var aPositionIndex = getInitialPosition($(a));
+                /** The initial index for b
+                 * @type {number} */
+                var bPositionIndex = getInitialPosition($(b));
+                if (aPositionIndex !== -1 && bPositionIndex !== -1) {
+                    return aPositionIndex < bPositionIndex
+                        ? COMPARISON_RESULTS.A_COMES_FIRST
+                        : COMPARISON_RESULTS.B_COMES_FIRST;
+                }
+                // If anything is unclear, do not change positions
+                return COMPARISON_RESULTS.DO_NOT_CHANGE;
+            }
+
+            /**
+             * Returns the initial position (index) inside the multiselect
+             * for the given option or optgroup
+             * @param {jQuery} $optionOrOptgroup - the option or optgroup with the stored position
+             * @returns {number} - the initial position of this element relative to the parent element
+             */
+            function getInitialPosition($optionOrOptgroup) {
+                if ($optionOrOptgroup !== undefined && $optionOrOptgroup.is("option,optgroup")) {
+                    return $optionOrOptgroup.data(DATA_POSITION);
+                }
+                return -1;
+            }
+
+            /**
+             * Stores the initial position (index) inside the multiselect
+             * for the given option or optgroup
+             * @param {jQuery} $optionOrOptgroup - the option or optgroup for which the position should be recorded
+             * @param {number} newPosition - the new stored position for this element
+             * @returns {jQuery} - returns the element for chaining
+             */
+            function setInitialPosition($optionOrOptgroup, newPosition) {
+                if ($optionOrOptgroup instanceof jQuery && $optionOrOptgroup.is("option,optgroup")) {
+                    $optionOrOptgroup.data(DATA_POSITION, newPosition);
+                }
+                return $optionOrOptgroup;
+            }
+
+            /**
+             * Either prepends the given search filter html to the target select element
+             * just uses an existing input for the filter. Also stores which
+             * selects should be affected by the filter.
+             * @param {jQuery} $targetSelect - the select element where the filter should be prepended
+             * @param {string|jQuery} searchFilterHtmlOrElement - the html for the search filter or an existing jQuery element
+             * @param {jQuery} $filteredSelects - the select elements that should be filtered with this input
+             * @returns {FilterRelation} - the filter/selects relation
+             */
+            function prependSearchFilter($targetSelect, searchFilterHtmlOrElement, $filteredSelects) {
```

Comment:
I allow the search option to provide either HTML like you or an existing jQuery element.
I also tell this function which select elements should be filtered by the input.